### PR TITLE
Custom script when Cassandra starts up

### PR DESCRIPTION
**Pull request by SuhailRahman**
| Q               | A
| --------------- | ---
| New feature?    | [X]
| License         | Apache 2.0

### What's in this PR?
 This PR has the extended functionality of running a custom script when the Cassandra pod is started up, using the postStart lifecycle hook. One use case is when the user wishes to create custom roles on the startup of Cassandra
### Why?
In case a user wishes to perform an operation immediately after Cassandra is started up, such as create a role with cqlsh, the script specified by /etc/cassandra_post_start.sh can be injected via configmap to run on post-start.

### CODE REVIEW

**Comment by ajaybhat** [comment]
@cscetbon can you please review this

File: pkg/controller/cassandracluster/generator.go [hunk 1]
```diff
@@ -752,6 +752,15 @@ func createCassandraContainer(cc *api.CassandraCluster, status *api.CassandraClu
 		},
 
 		Lifecycle: &v1.Lifecycle{
+		    PostStart: &v1.Handler{
+				Exec: &v1.ExecAction{
+					Command: []string{
+						"/bin/bash",
+						"-c",
+						"/etc/cassandra/cassandra_post_start.sh",
```
Comment (cscetbon) [inline pkg/controller/cassandracluster/generator.go:760]: Your changes would make any deployment that doesn't provide that script fail. Your PostStart action should check if the file exists and if it does should run.

Also it doesn't mean that cassandra was started but just that the container was started as it's mentioned in the documentation _"There is no guarantee, however, that the postStart handler is called before the Container's entrypoint is called"_
So I think there is still some work missing here

**Review by cscetbon — CHANGES_REQUESTED** [review]

File: pkg/controller/cassandracluster/generator.go (hunk 1 above)
Comment (cscetbon) [reply pkg/controller/cassandracluster/generator.go:760]: @ajaybhat @SuhailRahman 👆

**Comment by cscetbon** [comment]
Closing the PR for now as it's been opened for more than a month. Reopen it when you have more time to work on it. Thanks

**Pull request closed without merging**